### PR TITLE
migration: add relation when journal recid in publication info

### LINKED ISSUE #283: document: publication info - journal matching
Publication_info refers to two types of publications on CDS

A) record associated with journal which our library is subscribed to
B) NOT A)

in case A) the publication_info should be resolved as a series relation
in case B) publication_info stays as it was provided by cds-dojson

case A you can recognise via journal_record_legacy_recid field of migration

TASKS:
- [x] - map XML to JSON
`773__g` -> migration.journal_record_legacy_recid of the matching journal (has to be already migrated) 
- [x] create a relation of type serial between journal and the document (document should be of type periodical issue)
- [x] add `773__0` to ignored fields
- [x] clean the field `publication_info` of the records matched with a journal (also via script) as a redundant info)
- [x] review the rest of 773 fields (if implemented correctly, according to 
https://docs.google.com/spreadsheets/d/1OHMDiMfBD1Aam-kteIuBL9tEPK-C-Jevw-NcmGGxRvY/edit#gid=1257315908

### PR DESCRIPTION
**Pull request by KonstantinaStoikou**
Closes #283 
Closes https://github.com/orgs/inveniosoftware/projects/19#card-53138088

### CODE REVIEW

File: cds_ils/importer/providers/cds/ignore_fields.py [hunk 1]
```diff
@@ -68,6 +68,7 @@
     "695__a",
     "700__9",
     "710__5",
+    "773__0",
```
Comment (kprzerwa) [inline cds_ils/importer/providers/cds/ignore_fields.py:71]: could you add a comment ` # on library request ` ?

File: cds_ils/importer/providers/cds/rules/base.py [hunk 2]
```diff
@@ -245,28 +245,38 @@ def publication_info(self, key, value):
...
                 "year": clean_val("y", v, int),
             }
         )
 
+        rel_recid = clean_val("g", value, str)
+        if rel_recid:
+            recids = _migration.get("journal_record_legacy_recids", [])
+            recids.append({"recid": rel_recid, "volume": volume})
+            _migration["has_journal"] = True
+            self["document_type"] = "PERIODICAL_ISSUE"
```
Comment (kprzerwa) [inline cds_ils/importer/providers/cds/rules/base.py:270]: 👍

File: cds_ils/migrator/relations/api.py [hunk 3]
```diff
@@ -200,6 +201,23 @@ def link_records_and_serial(record_cls, search):
                     serial, record, SERIAL_RELATION, volume
                 )
 
+    def link_record_and_journal(record_cls, search):
```
Comment (kprzerwa) [inline cds_ils/migrator/relations/api.py:204]: let's add test for the linking?

Comment (KonstantinaStoikou) [reply cds_ils/migrator/relations/api.py:204]: I added them [here](https://github.com/CERNDocumentServer/cds-ils/blob/ac9f41bd3b0b24c000bf091d24e949ef5677b91d/tests/migrator/test_relations.py)

**Review by kprzerwa — APPROVED** [review]

**Pull request merged**